creator-node/src/components/replicaSet/URSMRegistrationComponentService.js

```diff
@@ -33,35 +36,16 @@ const { recoverWallet, signatureHasExpired } = require('../../apiSigning')
  * @param {string} reqSignature signature generated by requesting node, used in public key recovery
  */
 const respondToURSMRequestForSignature = async ({ libs: audiusLibs, nodeConfig }, logger, spID, reqTimestamp, reqSignature) => {
-  if (!spID || !reqTimestamp || !reqSignature) {
-    throw new ErrorBadRequest('Must provide all required query parameters: spID, timestamp, signature')
-  }
-
-  spID = parseInt(spID)
-
-  /**
-   * Fetch node info from L1 ServiceProviderFactory for spID
-   */
-  const spRecordFromSPFactory = await audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceEndpointInfo(
-    'content-node',
-    spID
-  )
-  let {
-    owner: ownerWalletFromSPFactory,
-    delegateOwnerWallet: delegateOwnerWalletFromSPFactory,
-    endpoint: nodeEndpointFromSPFactory
-  } = spRecordFromSPFactory
-  delegateOwnerWalletFromSPFactory = delegateOwnerWalletFromSPFactory.toLowerCase()
-
-  /**
-   * Reject if node is not registered as valid SP on L1 ServiceProviderFactory
-   */
-  if (
-    LibsUtils.isZeroAddress(ownerWalletFromSPFactory) ||
-    LibsUtils.isZeroAddress(delegateOwnerWalletFromSPFactory) ||
-    !nodeEndpointFromSPFactory
-  ) {
-    throw new ErrorBadRequest(`SpID ${spID} is not registered as valid SP on L1 ServiceProviderFactory`)
+  let ownerWalletFromSPFactory, delegateOwnerWalletFromSPFactory, nodeEndpointFromSPFactory
+  try {
+    ({ ownerWalletFromSPFactory, delegateOwnerWalletFromSPFactory, nodeEndpointFromSPFactory, spID } = await verifyRequesterIsValidSP({
```

Comment:
I merged these two functions into `verifyRequesterIsValidSP` because I thought that this entire functionality of getting the sp info and validating it should be in 1 fn that any caller can use, and not 2. this way, we can lessen the breadth of which fns a caller needs to make.

Reply:
yeah i know, i think it obfuscates unnecessarily but w/e, nbd